internal/graph/storagewrapper.go

```diff
@@ -203,7 +204,9 @@ func (c *CachedDatastore) newCachedIterator(
 
 	return &cachedIterator{
 		iter:          iter,
-		tuples:        make([]*openfgav1.Tuple, 0, c.maxResultSize),
+		relation:      relation,
```

Comment:
Can we abstract this away? Given how we can have multiple instances of an iterator? Maybe via a function?